pkg/reportingplugins/mercury/aggregate_functions.go

```diff
@@ -73,104 +72,40 @@ func GetConsensusAsk(paos []ParsedAttributedObservation, f int) (*big.Int, error
 	return validAsks[len(validAsks)/2], nil
 }
 
-type block struct {
-	hash string
-	num  int64
-	ts   uint64
-}
-
-func (b1 block) less(b2 block) bool {
-	if b1.num == b2.num && b1.ts == b2.ts {
-		// tie-break on hash, all else being equal
-		return b1.hash < b2.hash
-	} else if b1.num == b2.num {
-		// if block number is equal and timestamps differ, take the latest timestamp
-		return b1.ts > b2.ts
-	} else {
-		// if block number is different, take the higher block number
-		return b1.num > b2.num
-	}
-}
-
-// GetConsensusCurrentBlock gets the most common (mode) block hash/number/timestamps.
-// In the event of a tie, use the lowest numerical value
-func GetConsensusCurrentBlock(paos []ParsedAttributedObservation, f int) (hash []byte, num int64, ts uint64, err error) {
-	var validPaos []ParsedAttributedObservation
+// GetConsensusLinkFee gets the median link fee
+func GetConsensusLinkFee(paos []IParsedAttributedObservation, f int) (*big.Int, error) {
+	var validLinkFees []*big.Int
 	for _, pao := range paos {
-		if pao.CurrentBlockValid {
-			validPaos = append(validPaos, pao)
-		}
-	}
-	if len(validPaos) < f+1 {
-		return nil, 0, 0, fmt.Errorf("fewer than f+1 observations have a valid current block (got: %d/%d)", len(validPaos), len(paos))
-	}
-
-	m := map[block]int{}
-	maxCnt := 0
-	for _, pao := range validPaos {
-		b := block{string(pao.CurrentBlockHash), pao.CurrentBlockNum, pao.CurrentBlockTimestamp}
-		m[b]++
-		if cnt := m[b]; cnt > maxCnt {
-			maxCnt = cnt
+		fee, valid := pao.GetLinkFee()
+		if valid {
+			validLinkFees = append(validLinkFees, fee)
 		}
 	}
-
-	if maxCnt < f+1 {
-		return nil, 0, 0, errors.New("no unique block with at least f+1 votes")
-	}
-
-	var blocks []block
-	for b, cnt := range m {
-		if cnt == maxCnt {
-			blocks = append(blocks, b)
-		}
+	if len(validLinkFees) < f+1 {
+		return nil, errors.New("fewer than f+1 observations have a valid price")
 	}
-	sort.Slice(blocks, func(i, j int) bool {
-		return blocks[i].less(blocks[j])
+	sort.Slice(validLinkFees, func(i, j int) bool {
+		return validLinkFees[i].Cmp(validLinkFees[j]) < 0
 	})
 
-	return []byte(blocks[0].hash), blocks[0].num, blocks[0].ts, nil
+	return validLinkFees[len(validLinkFees)/2], nil
 }
 
-// GetConsensusMaxFinalizedBlockNum gets the most common (mode)
-// ConsensusMaxFinalizedBlockNum In the event of a tie, the lower number is
-// chosen
-func GetConsensusMaxFinalizedBlockNum(paos []ParsedAttributedObservation, f int) (int64, error) {
-	var validPaos []ParsedAttributedObservation
+// GetConsensusNativeFee gets the median native fee
+func GetConsensusNativeFee(paos []IParsedAttributedObservation, f int) (*big.Int, error) {
```

Comment:
Should this be moved to `v1/aggregate_functions.go`?

Reply:
This code will be shared between v1 and v2 so I'd keep it there for now